Extension/src/packageVersion.ts

```diff
@@ -0,0 +1,58 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+'use strict';
+
+export class PackageVersion {
+    major: number;
+    minor: number;
+    patch: number;
+    suffix?: string;
+    suffixVersion: number;
+
+    constructor(version: string) {
+        let tokens: string[] = version.split(new RegExp('[-\\.]', 'g')); // Match against dots and dashes
+        if (tokens.length < 3) {
+            throw new Error('Failed to parse version string: ' + version);
+        }
+
+        this.major = parseInt(tokens[0]);
+        this.minor = parseInt(tokens[1]);
+        this.patch = parseInt(tokens[2]);
+        this.suffix = tokens[3];
+        this.suffixVersion = tokens.length >= 5 ? parseInt(tokens[4]) : 0;
```

Comment:
I don't think this works with our naming conventions.  There is no `tokens[4]` because `insiders[N?]` doesn't split `insiders` from `N`.  (and there is no `insiders1`).  You need to remove `"insiders"` from tokens[3] and then parse the resulting integer, if present.